impeller/renderer/backend/vulkan/command_pool_vk.cc

```diff
@@ -148,6 +164,14 @@ void CommandPoolVK::GarbageCollectBuffersIfAble() {
   if (std::this_thread::get_id() != owner_id_) {
     return;
   }
+  for (auto& buffer : buffers_to_collect_) {
+    if (recycled_buffers_.size() >= kMaxRecycledBufferSize) {
+      // Don't grow boundlessly.
+      break;
+    }
+    buffer->reset();
+    recycled_buffers_.emplace_back(std::move(buffer));
```

Comment:
We might be able to get away with calling `reset` and `release`, letting the pool handle the recycling.

Reply:
I tried that and it didn't seem to be working as I expected.  Actually keeping the CommandBuffer around is a better way to assert that it is recycling at least.

Reply:
command buffers themselves don't own any memory, the command pool is where the pooled memory is.